### ISSUE TITLE
Hook not created in repository

### ISSUE THREAD

**Issue by s0undt3ch** [ericpaulbishop/redmine_git_hosting#30]
There should be a post-update hook in a newly created repository right?

```
# ls -lah /srv/vcs-repositories/repos-git/deluge/teste.git/
total 36K
drwxrwx--- 7 git git 4.0K 2011-07-12 12:28 .
drwxrwx--- 7 git git 4.0K 2011-07-12 12:28 ..
drwxrwx--- 2 git git 4.0K 2011-07-12 12:28 branches
-rw-rw---- 1 git git   66 2011-07-12 12:28 config
-rw-rw---- 1 git git   23 2011-07-12 12:28 HEAD
drwxrwx--- 2 git git 4.0K 2011-07-12 12:28 hooks
drwxrwx--- 2 git git 4.0K 2011-07-12 12:28 info
drwxrwx--- 4 git git 4.0K 2011-07-12 12:28 objects
drwxrwx--- 4 git git 4.0K 2011-07-12 12:28 refs
# ls -lah /srv/vcs-repositories/repos-git/deluge/teste.git/hooks/
total 56K
drwxrwx--- 2 git git 4.0K 2011-07-12 12:28 .
drwxrwx--- 7 git git 4.0K 2011-07-12 12:28 ..
-rwxrwx--- 1 git git  452 2011-07-12 12:28 applypatch-msg.sample
-rwxrwx--- 1 git git  894 2011-07-12 12:28 commit-msg.sample
lrwxrwxrwx 1 git git   54 2011-07-12 12:28 gitolite-hooked -> /usr/local/share/gitolite/hooks/common/gitolite-hooked
-rwxrwx--- 1 git git  160 2011-07-12 12:28 post-commit.sample
lrwxrwxrwx 1 git git   62 2011-07-12 12:28 post-receive.mirrorpush -> /usr/local/share/gitolite/hooks/common/post-receive.mirrorpush
-rwxrwx--- 1 git git  552 2011-07-12 12:28 post-receive.sample
-rwxrwx--- 1 git git  189 2011-07-12 12:28 post-update.sample
-rwxrwx--- 1 git git  398 2011-07-12 12:28 pre-applypatch.sample
-rwxrwx--- 1 git git 1.6K 2011-07-12 12:28 pre-commit.sample
-rwxrwx--- 1 git git 1.2K 2011-07-12 12:28 prepare-commit-msg.sample
-rwxrwx--- 1 git git 4.9K 2011-07-12 12:28 pre-rebase.sample
lrwxrwxrwx 1 git git   45 2011-07-12 12:28 update -> /usr/local/share/gitolite/hooks/common/update
-rwxrwx--- 1 git git 3.6K 2011-07-12 12:28 update.sample
```

**Comment by s0undt3ch** [comment]
And I'm still getting some:

```
fatal: /usr/lib/git-core/git-pull cannot be used without a working tree.
```

Didn't you addressed these issues?

**Comment by s0undt3ch** [comment]
Also testes with a parent'less test project, same non existing hook issue.

**Comment by ericpaulbishop** [comment]
Is this still broken after merging your latest fix (fixing the pull => fetch, merge in the second location) ?  Sorry I missed it in that second (more important) spot the first time around.

**Comment by s0undt3ch** [comment]
yes, notice the following:

```
www-data@ubxenweb:~$ sudo -u www-data ruby /srv/chili/projects/chiliproject/script/runner -e production "print \\\"good\\\""
Sorry, user www-data is not allowed to execute '/usr/bin/ruby /srv/chili/projects/chiliproject/script/runner -e production print \"good\"' as www-data on ubxenweb.ufsoft.org.
```

:)

**Comment by s0undt3ch** [comment]
The user running is already the web user, do you really want to sudo to him? specially since on current docs, the web user should only be allowed to sudo to the git user.

**Comment by ericpaulbishop** [comment]
That command should be getting run as the git user -- it's a test
whether the git user is correctly configured to run commands as the
www-data user.  The www-data user uses the run_as_git_user command to
run sudo BACK to the www-data user.  I suppose I could just remove the
test, since current configuration with sudo requires that git
repos/web server be in the same box, but previous ssh configuration
allowed the case where they were not (hence the reason for
testing...).

One reason I've left the test there is as a general test of whether
the run_as_git_user command is working properly -- I had some problems
until I realized that there are different versions of sudo that
actually behave differently (see my commits yesterday).  However,
since I've fixed that it seems to be working fine for me, so I'm not
sure why you're having problems.  If you comment out the test in the
git_hosting.rb file and just let the commands that set the post-update
hook run do those get called (and the hook set up) properly?

Eric

On Tue, Jul 12, 2011 at 10:16 AM, s0undt3ch
reply@reply.github.com
wrote:
